### PR TITLE
fix: Reset nonce as network is switched

### LINKED ISSUE #27788: Nonce does not update if we switch network between transactions
When switching network sometimes nonce is not reset

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Fix issue with nonce being not reset when switching network.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27788

## **Manual testing steps**

1. Go to test dapp
2. Switch network between submitting transactions
3. Ensure that nonce is correct each time

## **Screenshots/Recordings**
TODO

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/network-list-menu/network-list-menu.test.js [hunk 1]
```diff
@@ -19,11 +19,15 @@ const mockSetShowTestNetworks = jest.fn();
...
 const mockSetNetworkClientIdForDomain = jest.fn();
 const mockSetActiveNetwork = jest.fn();
+const mockUpdateCustomNonce = jest.fn();
+const mockSetNextNonce = jest.fn();
 
 jest.mock('../../../store/actions.ts', () => ({
   setShowTestNetworks: () => mockSetShowTestNetworks,
   setActiveNetwork: () => mockSetActiveNetwork,
   toggleNetworkMenu: () => mockToggleNetworkMenu,
+  updateCustomNonce: () => mockUpdateCustomNonce,
```
Comment (matthewwalsh0) [inline ui/components/multichain/network-list-menu/network-list-menu.test.js:29]: Minor, is it worth asserting these are called in a unit test?

File: ui/components/multichain/network-list-menu/network-list-menu.tsx [hunk 2]
```diff
@@ -277,6 +279,8 @@ export const NetworkListMenu = ({ onClose }: { onClose: () => void }) => {
             network.rpcEndpoints[network.defaultRpcEndpointIndex];
           dispatch(setActiveNetwork(networkClientId));
           dispatch(toggleNetworkMenu());
+          dispatch(updateCustomNonce(''));
```
Comment (matthewwalsh0) [inline ui/components/multichain/network-list-menu/network-list-menu.tsx:282]: Minor, is this handler big enough to warrant a `handleClick` lambda or function?

**Review by matthewwalsh0 — DISMISSED** [review]

File: ui/components/multichain/network-list-menu/network-list-menu.test.js (hunk 1 above)
Comment (jpuri) [reply ui/components/multichain/network-list-menu/network-list-menu.test.js:29]: Good point, I added that

File: ui/components/multichain/network-list-menu/network-list-menu.tsx (hunk 2 above)
Comment (jpuri) [reply ui/components/multichain/network-list-menu/network-list-menu.tsx:282]: I would prefer refactoring this in separate PR.

**Review by matthewwalsh0 — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Review by andreahaku — APPROVED** [review]
LGTM

**Pull request merged**